File: leetcode/3111-minimum-rectangles-to-cover-points/minRectanglesToCoverPoints.test.ts
import cases from "./cases.json";
import minRectanglesToCoverPoints from "./minRectanglesToCoverPoints";

test("minimum rectangles to cover points", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const result = minRectanglesToCoverPoints(testCase.input.points, testCase.input.w);
    console.timeEnd("time");
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});

File: leetcode/3111-minimum-rectangles-to-cover-points/minRectanglesToCoverPoints_test.rs
use std::{env::current_dir, path::Path};

use serde::Deserialize;

use crate::{util::load_cases, Solution};

#[derive(Deserialize, Debug)]
struct Input {
    points: Vec<Vec<i32>>,
    w: i32,
}

#[derive(Deserialize, Debug)]
struct Case {
    input: Input,
    output: i32,
}

#[test]
fn test_minimum_rectangles_to_cover_points() {
    let current_path = current_dir().unwrap().join(file!());
    let dir = current_path.parent().unwrap();

    let cases: Vec<Case> = load_cases(Path::new(&dir.join("./cases.json")));

    for case in &cases {
        assert_eq!(
            Solution::min_rectangles_to_cover_points(case.input.points.clone(), case.input.w),
            case.output
        );
    }
}
